tests/lib/rules/prefer-interface.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @fileoverview Prefer an interface declaration over a type literal (type T = { ... })
+ * @author Armano <https://github.com/armano2>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/prefer-interface"),
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+    parser: "typescript-eslint-parser",
+});
+
+ruleTester.run("interface-over-type-literal", rule, {
+    valid: [
+        `var foo = { };`,
+        `type U = string;`,
+        `type V = { x: number; } | { y: string; };`,
+        `
+type Record<T, U> = {
+    [K in T]: U;
+}
+        `,
```

Comment:
Should we `.trim()` these? I’ve been doing it for my PRs, but I don’t really mind either way.

Reply:
I really wish we could use something similar to what Prettier has here — plain `.ts` files with the input code, and a snapshot file containing all the error messages.

Reply:
we can but it's not following eslint guideline

Reply:
yeah but snapshotting might be a better solution :)

Reply:
we can go for it, but not in this PR xd

Reply:
than you should look on this; https://github.com/bradzacher/eslint-plugin-typescript/blob/master/tests/lib/rules/array-type.js#L783

Reply:
(⏩ #242)